test/denormalized_has_many_test.rb

```diff
@@ -88,4 +92,10 @@ def test_saving_associated_records_should_expire_itself_and_the_parents_cache
     IdentityCache.cache.expects(:delete).with(@record.primary_cache_index_key)
     child.save!
   end
+
+  def test_fetch_association_does_not_allow_chaining
```

Comment:
> I would prefer we test for the return type to be an `Array` than not to respond_to `where`.

The problem is that in rails 3.2 the collection proxy pretends to be an Array, so `record.associated_records.class` returns `Array` and `record.associated_records.class` returns `true`